### ISSUE TITLE
assume_integer_order in bessel.py should be more liberal

### ISSUE THREAD

**Issue by asmeurer** [sympy/sympy#11777]
Right now things like `jn(nu, x).rewrite(besselj)` won't work unless you explicitly set `nu = Symbol('nu', integer=True)`. It should be more liberal, only failing if the argument is not an integer. In general, SymPy should not require the correct assumptions to be true, only for the incorrect assumptions to not be True. 

As a side note: I'm unclear why this has to require nu be an integer. Aren't bessel functions defined for any value?

**Comment by parsoyaarihant** [comment]
Some tests are failing as the assumption of having an interger parameter is removed.
```python
assert jn(2.57082614100218 - 0.580811206370047*I, z) == sqrt(2)*sqrt(pi)*sqrt(1/z)*besselj(3.07082614100218 - 0.580811206370047*I, z)/2
```
Should I change the test in this case?

**Comment by asmeurer** [comment]
We should double check the result.

**Comment by yashsharan** [comment]
hello i would like to work on this. can you guide me where to start. I'm new here

**Comment by asmeurer** [comment]
I believe there is already a fix at https://github.com/sympy/sympy/pull/11785

**Comment by thisch** [comment]
@asmeurer, regarding your question about the integer order in the description of this issue, for spherical bessel functions (jn, yn) the order has to be an integer AFAIU. This is in contrast to normal bessel functions, where the order can even be complex-valued (modified bessel functions).

**Comment by asmeurer** [comment]
So is the merged PR #11785 incorrect?

**Comment by thisch** [comment]
I totally agree with what you said in https://github.com/sympy/sympy/issues/11777#issue-185453192:  "It should be more liberal, only failing if the argument is not an integer. In general, SymPy should not require the correct assumptions to be true, only for the incorrect assumptions to not be True." Since in PR #11785 there is no check for the incorrect assumptions to not be True, i.e., it does not check whether `nu != Symbol('nu', integer=False)`,  this PR does not address your statement. I'll work on a new PR to fix this.
